dandi/tests/fixtures.py

```diff
@@ -221,8 +233,56 @@ def fixture() -> Iterator[str]:
     return fixture
 
 
+def get_filtered_gitrepo_fixture(
+    url: str,
+    whitelist: List[str],
+) -> Callable[[], Iterator[str]]:
+    @pytest.fixture(scope="session")
+    def fixture() -> Iterator[str]:
+        skipif.no_network()
+        skipif.no_git()
+        path = tempfile.mktemp()  # not using pytest's tmpdir fixture to not
+        # collide in different scopes etc. But we
+        # would need to remove it ourselves
+        try:
+            lgr.debug("Cloning %r into %r", url, path)
+            runout = run(
+                [
+                    "git",
+                    "clone",
+                    "--depth=1",
+                    "--filter=blob:none",
+                    "--sparse",
+                    url,
+                    path,
+                ],
+                capture_output=True,
+            )
+            if runout.returncode:
+                raise RuntimeError(f"Failed to clone {url} into {path}")
+            # cwd specification is VERY important, not only to achieve the correct
+            # effects, but also to avoid dropping files from your repository if you
+            # were to run `git sparse-checkout` inside the software repo.
+            _ = run(["git", "sparse-checkout", "init", "--cone"], cwd=path)
+            _ = run(["git", "sparse-checkout", "set"] + whitelist, cwd=path)
+            yield path
+        finally:
+            try:
+                shutil.rmtree(path)
+            except BaseException as exc:
```

Comment:
DO NOT CATCH `BaseException`!  That catches keyboard interrupts and other things that should only be caught under very special circumstances.  Catch just `Exception` instead.